frontend/public/components/overview/resource-overview-page.tsx

```diff
@@ -9,19 +9,38 @@ import { NetworkingOverview } from './networking-overview';
 import { PodsOverview } from './pods-overview';
 import { resourceOverviewPages } from './resource-overview-pages';
 import { OverviewItem } from '@console/shared';
+import * as plugins from '../../plugins';
 
 const { common } = Kebab.factory;
 
-export const OverviewDetailsResourcesTab: React.SFC<OverviewDetailsResourcesTabProps> = ({
-  item: { buildConfigs, routes, services, pods, obj },
-}) => (
-  <div className="overview__sidebar-pane-body">
-    <PodsOverview pods={pods} obj={obj} />
-    <BuildOverview buildConfigs={buildConfigs} />
-    <NetworkingOverview services={services} routes={routes} />
-  </div>
+const getResourceTabSectionComp = (t) => (props) => (
+  <AsyncComponent {...props} loader={t.properties.loader} />
 );
 
+const getPluginTabSectionResource = (item) => {
+  const sections = plugins.registry
+    .getOverviewTabSections()
+    .filter((section) => item[section.properties.key]);
+  return sections.map((section) => getResourceTabSectionComp(section));
```

Comment:
You could also return a key based off of the [type](https://github.com/openshift/console/pull/3056/files#diff-34d17997d30d17137995c2847e41e305R37) that you could use in your `pluginComponents.map` call [below](https://github.com/openshift/console/pull/3056/files#diff-626eda35061076f284f6cec7908679f4R37).

```suggestion
  return sections.map((section) => ({
     Component: getResourceTabSectionComp(section),
     key: section.properties.key
  }));
```